airflow/www/templates/airflow/tree.html

```diff
@@ -108,6 +109,22 @@
 var diagonal = d3.svg.diagonal()
     .projection(function(d) { return [d.y, d.x]; });
 
+const taskTip = d3.tip()
+  .attr('class', 'tooltip')
+  .style('background', 'black')
+  .style('color', 'white')
+  .style("border", "solid")
+  .style("border-width", "1px")
+  .style("border-radius", "5px")
+  .style("padding", "10px")
+  .direction(function(toolTipHtml) {
```

Comment:
I don't like this - it feels error-prone. Can we instead to `tip.direction(condition ? 'e': 'n')` from inside the on mouseover event handler.

Reply:
Or, alternatively: we could do `taskTip.show(d, this)` (`d` here is the data/task struct from on-mouseover, and then have html/direction look at the object, not the string:

```js
   .direction(function(dl) {
    return d.depends_on_paste ? 'e' : 'n'
  })
  .html(function(d) {

        if (d.operator != undefined) {
          if (d.operator != undefined) {
            tt += "operator: " + escapeHtml(d.operator) + "<br/>";
          }
          tt += "depends_on_past: " + escapeHtml(d.depends_on_past) + "<br/>";
          tt += "upstream: " + escapeHtml(d.num_dep) + "<br/>";
          tt += "retries: " + escapeHtml(d.retries) + "<br/>";
          tt += "owner: " + escapeHtml(d.owner) + "<br/>";
          tt += "start_date: " + escapeHtml(d.start_date) + "<br/>";
          tt += "end_date: " + escapeHtml(d.end_date) + "<br/>";
        }
        return tt
});
```

(untested)

Reply:
that was my original design. but i did not like in the end as `d` is not always the object.

Reply:
love `tip.direction(condition ? 'e': 'n')`